internal/dynamo/client.go

```diff
@@ -188,10 +189,27 @@ func (c *Client) Put(ctx context.Context, v interface{}) error {
 		return err
 	}
 
-	_, err = c.svc.PutItem(ctx, &dynamodb.PutItemInput{
+	input := &dynamodb.PutItemInput{
 		TableName: aws.String(c.table),
 		Item:      item,
-	})
+	}
+
+	if updatedAt, exists := item["UpdatedAt"]; exists {
+		var updatedAtTime time.Time
+		err = attributevalue.Unmarshal(updatedAt, &updatedAtTime)
+		if err != nil {
+			return err
+		}
+
+		if !updatedAtTime.IsZero() {
+			input.ConditionExpression = aws.String("UpdatedAt < :updatedAt")
```

Comment:
Maybe it's too early for me to think, but I think this doesn't really change anything as it is "last write wins" which will still overwrite any other inflight writes